types/eslint/index.d.ts

```diff
@@ -276,6 +294,16 @@ export namespace SourceCode {
               filter?: FilterPredicate;
               count?: number;
           };
+
+    type TokenReturnType<
+        Opts extends CursorWithSkipOptions | CursorWithCountOptions | { includeComments?: boolean }
+    > = Opts extends (token: any) => token is infer T
+        ? T
+        : Opts extends { filter: (tokenOrComment: any) => tokenOrComment is infer T }
+        ? T
+        : Opts extends { includeComments: true }
```

Comment:
This should be `includeComments: boolean` since it is impossible to know statically that a call with `{ includeComments: foo }` will not produce a result containing comments.

Reply:
You're correct! I had thought of this, and thought I had tested it, but it must have been in an earlier draft. The PR is now updated to account for this and I added some tests to verify